### ISSUE TITLE
Tag DSLs don't allow suspending functions to be called

### ISSUE THREAD

**Issue by Tmpod** [ScottPierce/kotlin-html#29]
In the current version (`0.7.6`), coroutines can't be called inside HTML tag DSLs, something I rely on. In my opinion this is something that needs to be addressed as it severely limits the ability to use this library for dynamic page generation. Something like collecting items from a flow is impossible now.  
Here's an example:
```kotlin
val flow = ...
body {
    flow.collect {
        div {
            ...  // do stuff with the collected item 
        }
    }
}
```  
Inlining the DSLs would solve this issue.

**Comment by ScottPierce** [comment]
The DSL's is inline for performance reasons. The only elements that aren't inline are the void elements that don't take lambdas functions. Any function that doesn't take a lambda should already be inline. I tested this and was able to use coroutines properly inside the DSL, as long as the wrapping function was a suspend function.

**Comment by Tmpod** [comment]
Well that's strange. I can't provide a snippet of the exact code I'm running rn as I'm away from my PC but the weird part is that I didn't touch it and it stopped working when I upgraded, precisely because it wasn't able to run coroutines.

**Comment by ScottPierce** [comment]
A possible solution for this would be https://github.com/ScottPierce/kotlin-html/pull/33

Let's see if the tests pass, and I'll try merging it in. I can't remember why I chose for those not to be inlined.

**Comment by ScottPierce** [comment]
This should be resolved in `0.7.15`. Re-open this if you have any issue.
